### PR TITLE
feat(sampling): Rename `TransmissivityMatrix`

### PR DESCRIPTION
**Pull request by Kolarovszki**
`TransmissivityMatrix` got renamed to `LossyInterferometer`, since the
latter name is more descriptive regarding its usage.

### CODE REVIEW

**Review by Tomev — CHANGES_REQUESTED** [review]
Aside from the review comments I'd also like to discuss the 

test_LossyInterferometer_is_equivalent_to_Loss_and_Interferometers

test.

File: tests/backends/sampling/test_gates.py [hunk 1]
```diff
@@ -140,7 +140,7 @@ def test_lossy_program():
 
 
 @pytest.mark.monkey
-def test_TransmissivityMatrix(generate_unitary_matrix):
```
[the remark this answers is not in the document]
Comment (Kolarovszki) [reply tests/backends/sampling/test_gates.py:143]: True. Same should be done for the test above called `test_lossy_program`, right?

[the remark this answers is not in the document]
Comment (Kolarovszki) [reply tests/backends/sampling/test_gates.py:143]: done

File: piquasso/instructions/channels.py [hunk 2]
```diff
@@ -179,13 +179,13 @@ def _autoscale(self) -> None:
...
+class LossyInterferometer(Gate):
     """Applies a loss channel to the state via a transmissivity matrix.
 
     The transmissivity matrix :math:`A` should have singular values in the interval
     :math:`[0, 1]`.
 
-    Applying :class:`TransmissivityMatrix` with parameter :math:`A` on all modes is
+    Applying :class:`LossyInterferometer` with parameter :math:`A` on all modes is
     equivalent to the following::
         pq.Q() | pq.Interferometer(W) | pq.Loss(np.diag(D)) | pq.Interferometer(V)
```
[the remark this answers is not in the document]
Comment (Kolarovszki) [reply piquasso/instructions/channels.py:190]: done

[the remark this answers is not in the document]
Comment (Tomev) [reply piquasso/instructions/channels.py:190]: +1

File: piquasso/instructions/channels.py [hunk 3]
```diff
@@ -201,13 +201,13 @@ class TransmissivityMatrix(Gate):
             When the singular values are not in the interval :math:`[0, 1]`.
```
[the remark this answers is not in the document]
Comment (Tomev) [reply piquasso/instructions/channels.py:205]: Ah, yes, this part of the comment was skipped. I can see it now, it's good.

File: tests/backends/sampling/test_gates.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Tomev) [reply tests/backends/sampling/test_gates.py:143]: +1

**Review by Tomev — APPROVED** [review]
My only issue is that the tests don't pass, but I've been told that's not the issue with the code. In the light of that - I approve. :)

**Pull request merged**